plugins/stackstorm/src/components/ExecutionsTable/ExecutionsTable.tsx

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import { Link, Table, TableColumn } from '@backstage/core-components';
+import { useApi, configApiRef, errorApiRef } from '@backstage/core-plugin-api';
+import { Execution, stackStormApiRef } from '../../api';
+import { Status } from './Status';
+import { ExecutionPanel } from './ExecutionPanel';
+
+type DenseTableProps = {
+  executions: Execution[];
+  loading: boolean;
+  page: number;
+  pageSize: number;
+  onPageChange: (page: number) => void;
+  onRowsPerPageChange: (rows: number) => void;
+};
+
+export const DenseTable = ({
+  executions,
+  loading,
+  page,
+  pageSize,
+  onPageChange,
+  onRowsPerPageChange,
+}: DenseTableProps) => {
+  const config = useApi(configApiRef);
+
+  const columns: TableColumn<Execution>[] = [
+    {
+      title: 'Status',
+      field: 'status',
+      render: e => <Status status={e.status} />,
+    },
+    {
+      title: 'Time',
+      field: 'start_timestamp',
+      render: e => `${new Date(e.start_timestamp).toUTCString()}`,
+    },
+    { title: 'Name', field: 'action.ref' },
+    {
+      title: 'Execution ID',
+      field: 'id',
+      render: e => (
+        <Link
+          to={`${config.getString('stackstorm.webUrl')}/?#/history/${e.id}`}
+        >
+          {e.id}
+        </Link>
+      ),
+    },
+  ];
+
+  const count =
+    pageSize > executions.length
+      ? (page + 1) * pageSize + executions.length - pageSize
+      : (page + 1) * pageSize + 1;
+
+  return (
+    <Table
+      title="Executions"
+      columns={columns}
+      data={executions}
+      page={page}
+      totalCount={count}
+      isLoading={loading}
+      options={{
+        paging: true,
+        search: false,
+        pageSize: pageSize,
+        padding: 'dense',
+        showFirstLastPageButtons: false,
+      }}
+      onPageChange={onPageChange}
+      onRowsPerPageChange={onRowsPerPageChange}
+      detailPanel={rowData => {
+        return <ExecutionPanel id={rowData.rowData.id} />;
+      }}
+    />
+  );
+};
+
+export const ExecutionsTable = () => {
+  const st2 = useApi(stackStormApiRef);
+  const errorApi = useApi(errorApiRef);
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [data, setData] = useState<Execution[]>([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const getData = async () => {
+      setLoading(true);
+      await st2
+        .getExecutions(rowsPerPage, page * rowsPerPage)
+        .then(d => {
+          setData(d);
+        })
+        .catch(err => {
+          errorApi.post(err);
```

Comment:
Could this one use useAsync and rendering a `<ResponseErrorPanel error={error} />` on error, like the other views? The error reporting gets a bit inconsistent here. `useAsync` can also accept a dependencies array as its last argument just like useEffect, so that it issues updated requests as needed